### PR TITLE
Don't start watching heaters when target is 0

### PR DESCRIPTION
**Pull request by GMagician**
Implement #12438 request

Test needed. I've my printer busy for the next 7hr...

### CODE REVIEW

File: Marlin/src/module/temperature.cpp [hunk 1]
```diff
@@ -1436,7 +1436,7 @@ void Temperature::init() {
     #if HOTENDS == 1
       UNUSED(e);
     #endif
-    if (degHotend(HOTEND_INDEX) < degTargetHotend(HOTEND_INDEX) - (WATCH_TEMP_INCREASE + TEMP_HYSTERESIS + 1)) {
+    if (degHotend(HOTEND_INDEX) && degHotend(HOTEND_INDEX) < degTargetHotend(HOTEND_INDEX) - (WATCH_TEMP_INCREASE + TEMP_HYSTERESIS + 1)) {
```
Comment (thinkyhead) [inline Marlin/src/module/temperature.cpp]: Do the same for the bed, just below?

Comment (thinkyhead) [inline Marlin/src/module/temperature.cpp]: Is your hotend able to register a temperature of exactly 0°C? Maybe you meant…
```cpp
if (degTargetHotend(HOTEND_INDEX) && degHotend(HOTEND_INDEX) < degTargetHotend(HOTEND_INDEX) - (WATCH_TEMP_INCREASE + TEMP_HYSTERESIS + 1)) {
```

Comment (GMagician) [reply Marlin/src/module/temperature.cpp]: > Maybe you meant…

Yes thats' was my fault...

**Comment by GMagician** [comment]
@thinkyhead I didn't changed bed because I though this modify was helpful with fully detachable carriages (also electrical connection) but bed should always be present or absent. Why do you think also bed need such change?

**Comment by thinkyhead** [comment]
A detachable carriage does what? — Causes the thermistor to be disconnected? Thermistors register a random temperature (e.g., -14°C) when disconnected, not usually 0°C. If the intent is to ignore heater faults whenever a thermistor is detached, I'm not sure it's a good idea to apply it for all setups.

**Comment by GMagician** [comment]
No ok, my modify was wrong. I used bad function and your is right. I was talking of the same fix for bed

Edit: Why do you think bed needs such change?

**Comment by GMagician** [comment]
>  I'm not sure it's a good idea to apply it for all setups.

I'll add an option for it....Do you think it's better to write it in configuration.h or in configuration_adv.h?

**Comment by thinkyhead** [comment]
> Edit: Why do you think bed needs such change?

```cpp
#if HAS_HEATED_BED
  MENU_MULTIPLIER_ITEM_EDIT_CALLBACK(int3, MSG_BED, &thermalManager.target_temperature_bed, 0, BED_MAXTEMP - 15, watch_temp_callback_bed);
#endif
```

**Comment by thinkyhead** [comment]
> I'll add an option for it....

No need. Since the point is to ignore heaters with a target temperature set to 0, it makes sense based on the usage of the `start_watching_*` functions.

**Pull request merged**